pkg/virt-handler/network/cniproxy/cniproxy.go

```diff
@@ -0,0 +1,150 @@
+package cniproxy
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/golang/glog"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/utils"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+//TODO: Make it configurable
+const (
+	CNINetDir     = "/etc/cni/net.d"
+	CNIPluginsDir = "/opt/cni/bin"
+	LibvirtSocket = "/var/run/libvirt/libvirt-sock"
+)
+
+type CNIProxy struct {
+	cniConfig   *libcni.CNIConfig
+	netConfig   *libcni.NetworkConfig
+	runtimeConf *libcni.RuntimeConf
+}
+
+func confFiles(dir string, extensions []string) ([]string, error) {
+	// In part, adapted from rkt/networking/podenv.go#listFiles
+	files, err := ioutil.ReadDir(dir)
+
+	switch {
+	case err == nil: // break
+		glog.Errorln("err==nil")
+	case os.IsNotExist(err):
+		glog.Errorln("Doesnot exist: ", err)
+		return nil, nil
+	default:
+		glog.Errorln("default: ", err)
+		return nil, err
+	}
+
+	confFiles := []string{}
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
+		fileExt := filepath.Ext(f.Name())
+		for _, ext := range extensions {
+			glog.Errorln("match ext: ", ext)
+			if fileExt == ext {
+				glog.Errorln("adding file")
```

Comment:
Are these error messages for debugging? Some of them look informational/debug in nature.

Reply:
Yeah, that was just for debugging. It's not ready for a serious review yet :)